driver/src/it/scala/org/mongodb/scala/DocumentationExampleSpec.scala

```diff
@@ -0,0 +1,516 @@
+    )
+    // End Example 13
+
+    findObservable.execute().size should equal(2)
+  }
+
+  it should "be able to query embedded documents" in withCollection { collection =>
+
+    // Start Example 14
+    collection.insertMany(Seq(
+      Document("""{ item: "journal", qty: 25, size: { h: 14, w: 21, uom: "cm" }, status: "A" }"""),
+      Document("""{ item: "notebook", qty: 50, size: { h: 8.5, w: 11, uom: "in" }, status: "A" }"""),
+      Document("""{ item: "paper", qty: 100, size: { h: 8.5, w: 11, uom: "in" }, status: "D" }"""),
+      Document("""{ item: "planner", qty: 75, size: { h: 22.85, w: 30, uom: "cm" }, status: "D" }"""),
+      Document("""{ item: "postcard", qty: 45, size: { h: 10, w: 15.25, uom: "cm" }, status: "A" }""")
+    )).execute()
+    // End Example 14
+
+    collection.count().execute() should equal(5)
+
+    // Start Example 15
+    import org.mongodb.scala.model.Filters
+    var findObservable = collection.find(Filters.eq("size", Document("h" -> 14, "w" -> 21, "uom" -> "cm")))
+    // End Example 15
+
+    findObservable.execute().size should equal(1)
+
+    // Start Example 16
+    findObservable = collection.find(Filters.eq("size", Document("w" -> 21, "h" -> 14, "uom" -> "cm")))
+    // End Example 16
+
+    findObservable.execute().size should equal(0)
+
+    // Start Example 17
+    findObservable = collection.find(Filters.eq("size.uom", "in"))
+    // End Example 17
+
+    findObservable.execute().size should equal(2)
+
+    // Start Example 18
+    findObservable = collection.find(Filters.lt("size.h", 15))
+    // End Example 18
+
+    findObservable.execute().size should equal(4)
+
+    // Start Example 19
+    findObservable = collection.find(Filters.and(
+      Filters.lt("size.h", 15),
+      Filters.eq("size.uom", "in"),
+      Filters.eq("status", "D")
+    ))
+    // End Example 19
+
+    findObservable.execute().size should equal(1)
+  }
+
+  it should "be able to query array" in withCollection { collection =>
+
+    //Start Example 20
+    collection.insertMany(Seq(
+      Document("""{ item: "journal", qty: 25, tags: ["blank", "red"], dim_cm: [ 14, 21 ] }"""),
+      Document("""{ item: "notebook", qty: 50, tags: ["red", "blank"], dim_cm: [ 14, 21 ] }"""),
+      Document("""{ item: "paper", qty: 100, tags: ["red", "blank", "plain"], dim_cm: [ 14, 21 ] }"""),
+      Document("""{ item: "planner", qty: 75, tags: ["blank", "red"], dim_cm: [ 22.85, 30 ] }"""),
+      Document("""{ item: "postcard", qty: 45, tags: ["blue"], dim_cm: [ 10, 15.25 ] }""")
+    )).execute()
+    //End Example 20
+
+    collection.count().execute() should equal(5)
+
+    //Start Example 21
+    import org.mongodb.scala.model.Filters
+    var findObservable = collection.find(Filters.eq("tags", Seq("red", "blank")))
+    //End Example 21
+
+    findObservable.execute().size should equal(1)
+
+    //Start Example 22
+    findObservable = collection.find(Filters.all("tags", "red", "blank"))
```

Comment:
Acknowledged. In Scala varargs are strange and essentially a Seq, so you can't overload and you have to unpack if you want to use that syntax.

`Filters.all("tags", Seq("red", "blank"):_*))` doesn't read as well.